src/wxProfilerGUI/profilergui.h

```diff
@@ -66,24 +66,79 @@ struct AttachInfo
 
 wxBitmap LoadPngResource(const wchar_t *szName, const wxWindowBase* w);
 
+// Encapsulate a (config) value that can be overridden.
+//   `T` must be copyable and have a (sensible) default constructor.
+template<typename T>
+class OverridableOption
+{
+public:
+	OverridableOption()
+		: is_overridden(false)
+	{}
+	
+	OverridableOption( const T& config_value )
+		: config_value(config_value), is_overridden(false)
+	{}
+
+	// setters
+	void SetConfigValue( const T& new_value )
+	{
+		is_overridden = false;
+		config_value = new_value;
+	}
+	
+	void Override( const T& ovr_value )
+	{
+		is_overridden = true;
+		override_value = ovr_value;
+	}
+
+	// Using this replaces an override value.
+	T& operator =( const T& new_value )
+	{
+		is_overridden = false;
+		config_value = new_value;
+		return config_value;
+	}

```

Comment:
I think we need either this or `SetConfigValue`, not both. I would prefer the explicit method over the operator overload.